crates/common/src/config.rs

```diff
@@ -23,6 +23,39 @@ pub const CONTROL_ROOT: B256 =
 pub const BN254_CONTROL_ID: B256 =
     b256!("04446e66d300eb7fb45c9726bb53c793dda407a62e9601618bb43c5c14657ac0");
 
+/// Returns a value based on the provided `Option` and a default value, with safety checks.
```

Comment:
I am not a huge fan of this function:
- The naming is a bit confusing, only the default value is considered "unsafe" while it has nothing to do with the actual safety of the system. Can we maybe find a name that better reflects this?
- If I understand it correctly this function is only used to be able to have distinct serializations between None and Some. So maybe make sure that this function can only be used in this context.

Reply:
The entire concept might not be obvious to someone reading the config_hash function:
In config_hash and its helpers, the use of safe_default with values like u64::MAX means that explicitly setting these optional fields to those specific default values intended to cause an error will fail the hash generation. This is a specific design choice. Perhaps a comment explaining this behavior within the hashing functions could be beneficial for future clarity.